python/ray/tests/test_state_api.py

```diff
@@ -2341,58 +2373,6 @@ def _wait_to_start():
             e
         ), f"Expect an exception raised due to rate limit, but have {str(e)}"
 
-        # Kill the 3 slow running threads
```

Comment:
threading.Thread doesn't allow explicitly kill. So relying on timeout to simulate failure requests.